operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java

```diff
@@ -16,82 +16,162 @@
  */
 package org.keycloak.operator.v2alpha1;
 
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.EnvVarBuilder;
+import io.fabric8.kubernetes.api.model.HasMetadata;
 import io.fabric8.kubernetes.api.model.apps.Deployment;
 import io.fabric8.kubernetes.api.model.apps.DeploymentBuilder;
 import io.fabric8.kubernetes.client.KubernetesClient;
+import io.quarkus.logging.Log;
+import org.keycloak.operator.Config;
+import org.keycloak.operator.Constants;
+import org.keycloak.operator.OperatorManagedResource;
 import org.keycloak.operator.v2alpha1.crds.Keycloak;
-import org.keycloak.operator.v2alpha1.crds.KeycloakSpec;
-import org.keycloak.operator.v2alpha1.crds.KeycloakStatus;
+import org.keycloak.operator.v2alpha1.crds.KeycloakStatusBuilder;
 
 import java.net.URL;
+import java.util.HashMap;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
-import static org.keycloak.operator.v2alpha1.crds.KeycloakStatus.State.*;
+public class KeycloakDeployment extends OperatorManagedResource {
 
-public class KeycloakDeployment {
+//    public static final Pattern CONFIG_SECRET_PATTERN = Pattern.compile("^\\$\\{secret:([^:]+):(.+)}$");
 
-    KubernetesClient client = null;
+    private final Config config;
+    private final Keycloak keycloakCR;
+    private final Deployment existingDeployment;
+    private final Deployment baseDeployment;
 
-    KeycloakDeployment(KubernetesClient client) {
-        this.client = client;
+    public KeycloakDeployment(KubernetesClient client, Config config, Keycloak keycloakCR, Deployment existingDeployment) {
+        super(client, keycloakCR);
+        this.config = config;
+        this.keycloakCR = keycloakCR;
+
+        if (existingDeployment != null) {
+            Log.info("Existing Deployment provided by controller");
+            this.existingDeployment = existingDeployment;
+        }
+        else {
+            Log.info("Trying to fetch existing Deployment from the API");
+            this.existingDeployment = fetchExistingDeployment();
+        }
+
+        baseDeployment = createBaseDeployment();
     }
 
-    private Deployment baseDeployment;
+    @Override
+    protected HasMetadata getReconciledResource() {
```

Comment:
Probably using other variables names could make this less confusing ? wdyt ?

Reply:
Ok, unified the naming of the local variable with the field name.